static/js/src/PurchaseModal/components/Taxes/Taxes.tsx

```diff
@@ -48,10 +57,36 @@ const Taxes = ({ product, quantity }: TaxesProps) => {
     quantity,
     VATNumber: values.VATNumber,
   });
+  const genericPurchaseMutation = useMakePurchase();
 
   const onSaveClick = () => {
     setIsEditing(false);
-    taxMutation.mutate();
+    if (isGuest) {
+      taxMutation.mutate();
+    } else {
+      genericPurchaseMutation.mutate(
+        {
+          formData: values,
+          product,
+          quantity,
+          action: buyAction,
+          preview: true,
+        },
+        {
+          onSuccess: () => {
+            queryClient.invalidateQueries("preview");
+          },
+          onError: (error) => {
```

Comment:
If preview errors for example due to do not show up. If I insert random string in the VAT number field and hit save. `preview` errors due to invalid VAT but nothing surfaces to the user.

Reply:
Fixed :) 

<img width="858" alt="Screenshot 2023-01-04 at 4 21 41 pm" src="https://user-images.githubusercontent.com/57550290/210602777-feb2679b-51cb-4fb6-95a3-2a28fb91ea0c.png">